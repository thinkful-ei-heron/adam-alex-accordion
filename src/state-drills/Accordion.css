ul{
  list-style: none;
  width: 500px;
  text-align: center;
  margin: 0;
}

button{
  width: 100%;
  height: 33%;
}

p{
  border: 2px black solid;
  width: 100%;
  height: 100px;
  margin: 0;
  padding: 20%;
  padding-top: 10%;
}