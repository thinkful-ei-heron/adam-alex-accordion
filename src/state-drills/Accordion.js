import React, { Component } from 'react';
import './Accordion.css';

class Accordion extends React.Component {
  static defaultProps = {
    sections: []
  }
  
  constructor(props) {
    
    super(props)
    this.state={
      openItem: null
    }
  }

  componentWillUnmount() {
    clearInterval(this.interval)
  }

  handleButtonClick = (index) => {
    this.setState({
      openItem: index
    })
    if(this.state.openItem === index)
    {
      this.setState({
        openItem: null
      })
    }

  }
  render() {
    return (
      <ul className="accordion">
        {(this.props.sections !== []) ?
        (this.props.sections.map((e, index) => 
        <li className="section-component" key={ index } >
          <button className="section-button" id={ index } onClick={() => this.handleButtonClick(index)}>{e.title}</button>
          {this.state.openItem === index ? <p className="section-details">{this.props.sections[index].content}</p> : <l1></l1>}
        </li>)) : ''
        }
      </ul>
    )
  }
}

export default Accordion;