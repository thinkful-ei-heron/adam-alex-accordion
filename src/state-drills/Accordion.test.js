import { shallow } from 'enzyme'
import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { isTSAnyKeyword } from '@babel/types';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

//for no input
// it('renders empty given no sections', () => {
//   const wrapper = shallow(<Accordion />)
//   expect(toJson(wrapper)).toMatchSnapshot()
// })

it('opens a section when a button is clicked', () => {
  const wrapper = shallow(<Accordion sections={sections} />)
  wrapper.find('button').at(1).simulate('click')
  expect(toJson(wrapper)).toMatchSnapshot()
})

// describe('Accordion Test', () => {
//   // //smoke test
//   // it('render with no props', () => {
//   //   const div = document.createElement('div');
//   //   ReactDOM.render(<Accordion />, div);
//   //   ReactDOM.unmountComponentAtNode(div);
//   // })

//   // it('renders without open items', () => {
//   //   const tree = renderer
//   //     .create(<Accordion sections={sections} />)
//   //     .toJSON();
//   //     expect(tree).toMatchSnapshot();
//   // });

//   // it('renders an open item when clicked', () => {
//   //   const tree = renderer
//   //     .create(<Accordion sections={sections} />)
//   //     .toJSON();
//   //     expect(tree).toMatchSnapshot();
//   // })
  
// });