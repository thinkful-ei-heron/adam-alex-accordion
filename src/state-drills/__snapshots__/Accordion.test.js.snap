// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component only opens one section at a time 1`] = `
<ul
  className="accordion"
>
  <li
    className="section-component"
    key="0"
  >
    <button
      className="section-button"
      id={0}
      onClick={[Function]}
    >
      Section 1
    </button>
    <l1 />
  </li>
  <li
    className="section-component"
    key="1"
  >
    <button
      className="section-button"
      id={1}
      onClick={[Function]}
    >
      Section 2
    </button>
    <l1 />
  </li>
  <li
    className="section-component"
    key="2"
  >
    <button
      className="section-button"
      id={2}
      onClick={[Function]}
    >
      Section 3
    </button>
    <l1 />
  </li>
</ul>
`;

exports[`Accordion Component opens any clicked section 1`] = `
<ul
  className="accordion"
>
  <li
    className="section-component"
    key="0"
  >
    <button
      className="section-button"
      id={0}
      onClick={[Function]}
    >
      Section 1
    </button>
    <l1 />
  </li>
  <li
    className="section-component"
    key="1"
  >
    <button
      className="section-button"
      id={1}
      onClick={[Function]}
    >
      Section 2
    </button>
    <l1 />
  </li>
  <li
    className="section-component"
    key="2"
  >
    <button
      className="section-button"
      id={2}
      onClick={[Function]}
    >
      Section 3
    </button>
    <l1 />
  </li>
</ul>
`;

exports[`Accordion Component renders empty given no Accordion without errors 1`] = `
<ul
  className="accordion"
/>
`;

exports[`Accordion Component renders the no sections by default 1`] = `
<ul
  className="accordion"
>
  <li
    className="section-component"
    key="0"
  >
    <button
      className="section-button"
      id={0}
      onClick={[Function]}
    >
      Section 1
    </button>
    <l1 />
  </li>
  <li
    className="section-component"
    key="1"
  >
    <button
      className="section-button"
      id={1}
      onClick={[Function]}
    >
      Section 2
    </button>
    <l1 />
  </li>
  <li
    className="section-component"
    key="2"
  >
    <button
      className="section-button"
      id={2}
      onClick={[Function]}
    >
      Section 3
    </button>
    <l1 />
  </li>
</ul>
`;
